### PR TITLE
(#598) Promoblock With/Without Image

### LINKED ISSUE #598: Build: Promo Block -  F. Promo Block Dark
### Build Promo Block - Dark Text Box variation of promo block component based on design specs/requirements documented in this ticket

> **ESTIMATE** TBD

### Styling: 
*Anatomy of a component*
- [ ] Use correct spacing (8px grid) size, emphasis, alignment

### Behaviors
- Promo Block title on Widescreen & Desktop should be 3XL
- Promo Block title on Tablet Large, Tablet & Mobile Large should be 2XL
- Promo Block title on Mobile should be XL

### InVision: 
  * Widescreen mockup: https://publicis.invisionapp.com/console/MVP2-R2.1-Landing-Pages-cl756hway0281013t70e36jn0/cl9ftyfm40594017n0wlv2cg1/play
  * Desktop mockup: https://publicis.invisionapp.com/console/MVP2-R2.1-Landing-Pages-cl756hway0281013t70e36jn0/cl9ftyfm40595017ncuofhrc6/play
  * Tablet Large mockup: https://publicis.invisionapp.com/console/MVP2-R2.1-Landing-Pages-cl756hway0281013t70e36jn0/cl9ftyfm40596017ne0xu72cu/play
  * Tablet mockup: https://publicis.invisionapp.com/console/MVP2-R2.1-Landing-Pages-cl756hway0281013t70e36jn0/cl9ftyfm40597017ndfyq71j5/play
  * Mobile Large mockup: https://publicis.invisionapp.com/console/MVP2-R2.1-Landing-Pages-cl756hway0281013t70e36jn0/cl9ftyfm40598017n4qeid6e0/play
  * Mobile mockup: https://publicis.invisionapp.com/console/MVP2-R2.1-Landing-Pages-cl756hway0281013t70e36jn0/cl9ftyfm40599017ngry973ak/play

### Acceptance Criteria for Behaviors 
*If a component changes between breakpoints, or things show/hide, that gets captured here* 

```gherkin
As a user, I would like to see a promo block so that I can see the promoted content easily. 

Scenario: User views a page with a promo block with dark text box on Widescreen
Given a user is viewing a landing page with a promo block
  And the browser width is 1400px or larger  
Then a promo block will be displayed
  And the promo block will display with a dark text box
  And the title will be displayed
  And the title will have spacing to the top of the container with a spacing unit token of 10
  And the title will be center aligned to the width of the promo block
  And the title will have spacing to the left and right of the container with a spacing unit token of 6
  And the title will be font size theme token 3XL
  And if the description is set, it will be displayed below the title with a spacing unit token of 3
  And if the description is set, it will have spacing to the left and right of the container with a spacing unit token of 6
  And if the description is set, it will be font size theme token ???
  And the button will be displayed below the title or description with a spacing unit token of 3
  And the button will be center aligned to the width of the promo block
  And the button will have spacing to the bottom of the container with a spacing unit token of 10
  And the button will have the secondary button style
  And on hover, the buttons will appear with the secondary button hover style 
  And clicking on the button will take the user to the corresponding page
  
Scenario: User views a page with a promo block with dark text box on Desktop
Given a user is viewing a landing page with a promo block
  And the browser width is between 1024px and 1399px  
Then a promo block will be displayed
  And the promo block will display with a dark text box
  And the title will be displayed
  And the title will have spacing to the top of the container with a spacing unit token of 10
  And the title will be center aligned to the width of the promo block
  And the title will have spacing to the left and right of the container with a spacing unit token of 10
  And the title will be font size theme token 3XL
  And if the description is set, it will be displayed below the title with a spacing unit token of 2
  And if the description is set, it will have spacing to the left and right of the container with a spacing unit token of 10
  And if the description is set, it will be font size theme token ???
  And the button will be displayed below the title or description with a space equal to the spacing between the title and description (spacing token of 2 if there is no description)
  And the button will be center aligned to the width of the promo block
  And the button will have spacing to the bottom of the container with a spacing unit token of 10
  And the button will have the secondary button style
  And on hover, the buttons will appear with the secondary button hover style 
  And clicking on the button will take the user to the corresponding page

Scenario: User views a page with a promo block with dark text box on Tablet Large
Given a user is viewing a landing page with a promo block
  And the browser width is between 880px and 1023px  
Then a promo block will be displayed
  And the promo block will display with a dark text box
  And the title will be displayed
  And the title will have spacing to the top of the container with a spacing unit token of 9
  And the title will be center aligned to the width of the promo block
  And the title will have spacing to the left and right of the container with a spacing unit token of 8
  And the title will be font size theme token 2XL
  And if the description is set, it will be displayed below the title with a spacing unit token of 2
  And if the description is set, it will have spacing to the left and right of the container with a spacing unit token of 8
  And if the description is set, it will be font size theme token ???
  And the button will be displayed below the title or description with a space equal to the space between the title and description (spacing unit token of 2)
  And the button will be center aligned to the width of the promo block
  And the button will have spacing to the bottom of the container with a spacing unit token of 9
  And the button will have the secondary button style
  And on hover, the buttons will appear with the secondary button hover style 
  And clicking on the button will take the user to the corresponding page

Scenario: User views a page with a promo block with dark text box on Tablet 
Given a user is viewing a landing page with a promo block
  And the browser width is between 640px and 879px
Then a promo block will be displayed
  And the promo block will display with a dark text box
  And the title will be displayed
  And the title will have spacing to the top of the container with a spacing unit token of 9
  And the title will be center aligned to the width of the promo block
  And the title will have spacing to the left and right of the container with a spacing unit token of 5
  And the title will be font size theme token 2XL
  And if the description is set, it will be displayed below the title with a spacing unit token of 2
  And if the description is set, it will have spacing to the left and right of the container with a spacing unit token of 5
  And if the description is set, it will be font size theme token ???
  And the button will be displayed below the title or description with a space equal to distance between the title and description (spacing unit token of 2)
  And the button will be center aligned to the width of the promo block
  And the button will have spacing to the bottom of the container with a spacing unit token of 9
  And the button will have the secondary button style
  And on hover, the buttons will appear with the secondary button hover style 
  And clicking on the button will take the user to the corresponding page

Scenario: User views a page with a promo block with dark text box on Mobile Large
Given a user is viewing a landing page with a promo block
  And the browser width is between 480px and 639px
Then a promo block will be displayed
  And the promo block will display with a dark text box
  And the title will be displayed
  And the title will have spacing to the top of the container with a spacing unit token of 4
  And the title will be center aligned to the width of the promo block
  And the title will have spacing to the left and right of the container with a spacing unit token of 4
  And the title will be font size theme token 2XL
  And if the description is set, it will be displayed below the title with a spacing unit token of 2
  And if the description is set, it will have spacing to the left and right of the container with a spacing unit token of 4
  And if the description is set, it will be font size theme token ???
  And the button will be displayed below the title or description with a space equal to the space between the title and description (spacing unit token of 2)
  And the button will be center aligned to the width of the promo block
  And the button will have spacing to the bottom of the container with a spacing unit token of 4
  And the button will have the secondary button style
  And on hover, the buttons will appear with the secondary button hover style 
  And clicking on the button will take the user to the corresponding page

Scenario: User views a page with a promo block with dark text box on Mobile 
Given a user is viewing a landing page with a promo block
  And the browser width is 479px or smaller
Then a promo block will be displayed
  And the promo block will display with a dark text box
  And the title will be displayed
  And the title will have spacing to the top of the container with a spacing unit token of 4
  And the title will be center aligned to the width of the promo block
  And the title will have spacing to the left and right of the container with a spacing unit token of 4
  And the title will be font size theme token XL
  And if the description is set, it will be displayed below the title with a spacing unit token of 2
  And if the description is set, it will have spacing to the left and right of the container with a spacing unit token of 4
  And if the description is set, it will be font size theme token ???
  And the button will be displayed below the title or description with a spacing unit token of 3
  And the button will be center aligned to the width of the promo block
  And the button will have spacing to the bottom of the container with a spacing unit token of 4
  And the button will have the secondary button style
  And on hover, the buttons will appear with the secondary button hover style 
  And clicking on the button will take the user to the corresponding page
```

---
### Scenarios
**Expected use case:**
- Title: NCI Annual Plan & Budget Proposal for Fiscal Year 2024
- Description: NCI's research strategy supports investigator-initiated research and maximizes opportunities in emerging areas of science. The FY 2024 Annual Plan & Budget Proposal aligns with NCI’s vision and supports cancer research.
- Button: Learn More

**Long title/button edge case:**
- Title: Study identifies novel genomic changes in the most common type of lung cancer; TCGA finds mutations in a key cancer-causing pathway, expanding targets for existing drugs
- Description: Researchers from The Cancer Genome Atlas (TCGA) Research Network have identified novel mutations in a well-known cancer-causing pathway in lung adenocarcinoma, the most common subtype of lung cancer.
- Button: Click Here to Learn More and to Learn More Click Here

**Long description edge case:**
- Title: NCI Center to Reduce Cancer Health Disparities (CRCHD)
- Description: The Center to Reduce Cancer Health Disparities (CRCHD) is central to NCI’s efforts to reduce the unequal burden of cancer in our society via basic and community research, as well as networks, and to train the next generation of competitive researchers from diverse populations in cancer and cancer health disparities research.
- Button: Learn More

**All-around bad edge case:**
- Title: Un nuevo análisis de los subtipos de cáncer de seno podría llevar a una mejor estratificación del riesgo; Informe Anual a la Nación indica que la mortalidad y la incidencia de la mayoría de los cánceres continúan disminuyendo
- Description: Por primera vez, los investigadores han usado datos a nivel nacional para determinar la incidencia de los cuatro subtipos moleculares principales de cáncer de seno (mama) por edad, raza y grupo étnico, nivel de pobreza y varios otros factores. El informe mostró una continua reducción en las muertes por cáncer tanto en hombres, mujeres y niños, y para los cánceres que afectan a casi todos los lugares principales del cuerpo.
- Button: 

### Prerequisites
- *required dependency for identifying the problem, scope and then solution for this issue*

- [ ] Short Spike Description - Issue #9999

### Tasks
- [ ] Update Documentation Site (docs)
  - [ ] Create or update [INSERT DOCUMENTATION PAGE OR SECTION ON PAGE HERE] that includes:
    - Developer Documentation
      -  What SASS partial to import
      - (HTML Tab) What is the HTML?
      - (HTML Tab) (for vanilla js) What is the js library to import?
      - (React Tab) What is the react library to import?
      - (React Tab) What is the react markup?
    - **IA Provided** Guidance Documentation  
      - Includes when/how the component should be used
  - [ ] If the component is a global component like the footer, then it needs to be added to the documentation site as well as a real usage.
- [ ] Update NCIDS-CSS
  - **Insert list here of partials to be created** (There should be a sass partial available for all variants that are being created in this ticket)
- [ ] Update NCIDS-JS
  - **Insert requirements/list of components for JS**
- [ ] Update NCIDS-React
  - [ ] **Insert requirements/list of components for react**
    - (it can also be a separate ticket, should be linked here)

### PR DESCRIPTION
**Pull request by dlescarbeau**
Closes #598

- Adds Promoblock and alternating Promoblocks to CGov Home example page

### CODE REVIEW

File: packages/ncids-css/scss/settings/_settings-utilities.scss [hunk 1]
```diff
@@ -41,8 +41,8 @@ $theme-utility-breakpoints: (
 	'mobile-lg': true,
 	// 640px:
 	'tablet': true,
-	// 800px:
-	'tablet-lg': false,
+	// 880px:
+	'tablet-lg': true,
```
Comment (bryanpizzillo) [inline packages/ncids-css/scss/settings/_settings-utilities.scss:45]: This should be reverted an moved to the index.scss for the example site. Adding tablet-lg blows up the utilities. We don't care about the CSS size for example site or storybook, so we should only turn on utilities/expand the size of utilities in those things that specifically need them.

**Review by bryanpizzillo — CHANGES_REQUESTED** [review]
Based on the 1/12/2022 design meeting the left and right padding on the promo blocks should be the same between mobile and tablet-lg. The padding of the default (i.e. the one without an image) is the correct padding. 

There was another issue discovered that the heading weight was 700. It should be 600, and that would come from settings. (e.g. `$theme-font-weight-bold: 600 !default;`) This is probably not isolated to the promoblocks unless it is some mixin usage we had not done before.

------

To bad PR request changes do not have a nice resolve button. Anyway this was is fixed in the current version.

File: packages/ncids-css/scss/settings/_settings-utilities.scss (hunk 1 above)
Comment (bryanpizzillo) [reply packages/ncids-css/scss/settings/_settings-utilities.scss]: This has been changed.

**Review by sarinapadilla — APPROVED** [review]
Product approved!!

**Review by bryanpizzillo — APPROVED** [review]

**Pull request merged**